### PR TITLE
Fix issue when no child categories are present

### PR DESCRIPTION
**Pull request by darokel**
Fixes recent issue of category pages blowing up and returning 500s. Incident report can be found here (if you're a MAS employee) https://moneyadviceserviceuk.atlassian.net/wiki/display/DEV/2015-04-23+-+Site+Category+Pages+Down.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -108,7 +108,9 @@ def corporate_categories
   def corporate_category?(category, corporate = corporate_categories, categories = [])
     categories << corporate.map {|c| c['id']}
     return true if categories.flatten.include?(category)
-    corporate_category?(category, corporate.first['contents'], categories.flatten) if corporate.first['contents']
+    unless corporate.first['contents']
```
Comment (asmega) [inline app/controllers/application_controller.rb]: how does this fix the problem?

Comment (darokel) [inline app/controllers/application_controller.rb]: Checks for empty child categories first before trying to run stuff on `nil`.

Comment (asmega) [inline app/controllers/application_controller.rb]: ah I think I remember now if has empty contents if it has children.
would `if corporate.has_children?` not be better?

Comment (darokel) [inline app/controllers/application_controller.rb]: It would be, but right here corporate is just an `Array` that's put together from another method.

**Comment by asmega** [comment]
:+1:

**Pull request merged**